contrib/initramfs/Makefile.am

```diff
@@ -2,10 +2,12 @@ initrddir = $(datarootdir)/initramfs-tools
 
 initrd_SCRIPTS = conf-hooks.d/zfs hooks/zfs scripts/zfs scripts/local-top/zfs
 
+SUBDIRS = hooks scripts
+
 EXTRA_DIST = \
 	$(top_srcdir)/contrib/initramfs/conf-hooks.d/zfs \
-	$(top_srcdir)/contrib/initramfs/hooks/zfs \
-	$(top_srcdir)/contrib/initramfs/scripts/zfs \
+	$(top_srcdir)/contrib/initramfs/hooks/zfs.in \
+	$(top_srcdir)/contrib/initramfs/scripts/zfs.in \
```

Comment:
Since you've added proper Makefile's for the `hooks` and `scripts` sub-directories you can do the EXTRA_DIST for the `.in` file in each respective Makefile.  They should be removed entirely from here.  You can use `make dist` and check the tarball as a way to verify they were properly included.